File: targets/python-web/vulns/vuln18_token.py
import random
from flask import Blueprint, request
bp = Blueprint("vuln18", __name__)

@bp.get("/vuln18/token")
def token():
    user = request.args.get("user", "")
    # VULN:VULN-18:CWE-330:pattern predictable token from a seedable, non-CSPRNG generator
    rng = random.Random(hash(user))
    return "".join("%x" % rng.randint(0, 15) for _ in range(16))

File: targets/jsts-web/src/vulns/vuln18_token.ts
import { Router } from "express";
export const router = Router();
router.get("/vuln18/token", (req, res) => {
  // VULN:VULN-18:CWE-330:pattern predictable token from Math.random (not cryptographically secure)
  let t = "";
  for (let i = 0; i < 16; i++) t += Math.floor(Math.random() * 16).toString(16);
  res.send(t);
});
